src/rust/engine/fs/store/src/lib.rs

```diff
@@ -1464,9 +1463,19 @@ impl Store {
     // It also has the benefit of playing nicely with Docker for macOS file virtualization: see
     // #18162.
     #[cfg(target_os = "macos")]
-    copy(target, destination).await?;
+    copy(&target, &destination).await.map_err(|e| {
+      format!(
+        "Failed to copy from {target} to {}: {e}",
+        destination.display()
+      )
+    })?;
     #[cfg(not(target_os = "macos"))]
-    hard_link(target, destination).await?;
+    hard_link(&target, &destination).await.map_err(|e| {
+      format!(
+        "Failed to create hardlink to {target} at {}: {e}",
```

Comment:
Can you not use `destination:?` in the braces?

Reply:
No. `Path` and `PathBuf` don't actually implement `Display`: only `Debug`. To get `Display`, you have to ask for it explicitly. https://stackoverflow.com/a/24061240

Reply:
Huh